client/src/components/shared/GeneralPagination/index.js

```diff
@@ -0,0 +1,73 @@
+// @flow
```

Comment:
rebase and use the pagination lib from master, should be able to do all the stuff needed